front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -74,35 +79,30 @@ const TimesStops = ({
       className="time-stops-datasheet"
       columns={columns}
       value={rows}
-      onChange={(row: PathWaypointRow[], [op]) => {
+      onChange={(newRows: PathWaypointRow[], [op]) => {
         if (!isInputTable) {
           return;
         }
-        const rowData = { ...row[op.fromRowIndex] };
-        const previousRowData = rows[op.fromRowIndex];
-        if (
-          rowData.departure &&
-          rowData.arrival &&
-          (rowData.arrival !== previousRowData.arrival ||
-            rowData.departure !== previousRowData.departure)
-        ) {
-          rowData.stopFor = String(time2sec(rowData.departure) - time2sec(rowData.arrival));
-        }
-        if (!rowData.stopFor && op.fromRowIndex !== allWaypoints.length - 1) {
-          rowData.onStopSignal = false;
-        }
-        if (rowData.theoreticalMargin && !marginRegExValidation.test(rowData.theoreticalMargin)) {
-          rowData.isMarginValid = false;
-          setRows(row);
+        let updatedRows = [...newRows];
+        updatedRows[op.fromRowIndex] = updateRowTimesAndMargin(
+          newRows[op.fromRowIndex],
+          rows[op.fromRowIndex],
+          op,
+          allWaypoints.length
+        );
+        updatedRows = updateDaySinceDeparture(updatedRows, startTime);
+        if (!updatedRows[op.fromRowIndex].isMarginValid) {
+          setRows(newRows);
         } else {
-          rowData.isMarginValid = true;
-          if (op.fromRowIndex === 0) {
-            rowData.arrival = null;
-            // As we put 0% by default for origin's margin, if the user removes a margin without
-            // replacing it to 0% (undefined), we change it to 0%
-            if (!rowData.theoreticalMargin) rowData.theoreticalMargin = '0%';
-          }
-          dispatch(upsertViaFromSuggestedOP(rowData as SuggestedOP));
+          const newVias = updatedRows
+            .filter((row, index) => !isEqual(row, rows[index]))
+            .map((row) => {
+              return {
+                ...row,
+                ...(row.arrival && { arrival: durationSinceStartTime(startTime, row.arrival) }),
+              } as SuggestedOP;
+            });
```

Comment:
To remove the eslint warning

```suggestion
            .map(
              (row) =>
                ({
                  ...row,
                  ...(row.arrival && { arrival: durationSinceStartTime(startTime, row.arrival) }),
                }) as SuggestedOP
            );
```

Reply:
done